src/app/util/icon/icon.component.ts

```diff
@@ -9,25 +9,23 @@ import { DB_Icons } from '../../cdclient';
   styleUrls: ['./icon.component.css']
 })
 export class IconComponent implements OnInit {
-
-  @Input() iconID: number;
+  private _iconID: number;
   @Input() width: string = '64px';
   @Input() caption: boolean = true;
   icon: DB_Icons;
 
-  constructor(private luJsonService: LuJsonService) {
-  }
+  get iconID(): number { return this._iconID; }
 
-  ngOnInit() {
-  	this.getIcon();
+  @Input() set iconID(value: number) {
+    this._iconID = value;
+    if (this.iconID != undefined) {
+      this.luJsonService.getIcon(this.iconID).subscribe(icon => this.icon = icon);
```

Comment:
Setting local values in subscribe like this seems to be not ideal for change detection to pick up, at least in theory and tests. Ideally it would be a
```
$icon: Observable<DB_Icons>;
```
that is initialized using `$icon = $id.pipe(switchMap(id => luJson.get(id))` where $id is a `Subject` that you update with the setter or so.

But that's not what you're changing here so we can leave that for another PR if you want